packages/next-dashboard/src/components/layout/Main.js

```diff
@@ -61,75 +62,86 @@ function DashboardLayout({
       window.removeEventListener('resize', noTransition);
     };
   }, []);
+  const context = useContext(DashboardContext);
+  if (!context) {
+    throw new TypeError('Dashboard Layout needs to be in a Dashboard Context');
+  }
+
+  const {
+    getState,
+    theme: { class: themeClass },
+    modalActive,
+  } = context;
+
+  const sidebarActive = getState<boolean>('sidebarActive', false);
+  const sidebarCompact = getState<boolean>('sidebarCompact', false);
+
+  const toggleSidebarActive = () => {
+    context.setState<boolean>('sidebarActive', !sidebarActive);
+  };
+
+  const [scrollOffset, setScrollOffset] = useState<number | null>(null);
+  const [savedScrollOffset, setSavedScrollOffset] = useState<number | null>(
+    null,
+  );
+
+  // When a modal is shown, we want to save the top offset and set position fixed to disable scrolling
+  useEffect(() => {
+    if (scrollOffset === null && !modalActive) return;
+    if (modalActive) setScrollOffset(window.pageYOffset);
+    else {
+      setSavedScrollOffset(scrollOffset);
+      setScrollOffset(null);
+    }
+  }, [modalActive]);
+
+  // Once the modal isn't showing anymore, we want to scroll back to the old position, AFTER the DOM
+  // has updated with the new regular positioning.
+  useEffect(() => {
+    if (savedScrollOffset) window.scrollTo(0, savedScrollOffset);
+    setSavedScrollOffset(null);
+  }, [savedScrollOffset]);
```

Comment:
I wonder if it'd be reasonable to extract out the "scrollbar hack" management into its own hook, since it does seem like its own fairly orthogonal, separate thing that you might want to reuse/add to other components as well. Thoughts?

Reply:
Hmm, sounds reasonable, what should the hook return?

An object with `scrollOffset`, and and a boolean `shouldFix`

Reply:
That'd be one design, sure. I don't have a firm opinion here really, I don't have the problem loaded into my head :p but whatever you think works well

Reply:
Hmm I have another idea, let's see if you like it